src/components/EarnReport.tsx

```diff
@@ -8,172 +12,376 @@ import Balance from './Balance'
+  feeInSats: 'FEE_IN_SATS',
+  earnedAmountInSats: 'EARNED_AMOUNT_IN_SATS',
 }
 
-const YieldgenReportTable = ({ lines, maxAmountOfRows = 15 }: YielgenReportTableProps) => {
-  const { t } = useTranslation()
-  const settings = useSettings()
+const TABLE_THEME = {
+  Table: `
+    --data-table-library_grid-template-columns: 2fr 2fr 1fr 2fr 2fr 2fr 2fr;
+    font-size: 0.9rem;
+  `,
+  BaseCell: `
+    &:nth-of-type(2) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(3) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(4) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(5) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(6) button {
+      display: flex;
+      justify-content: end;
+    }
+  `,
+  Cell: `
+    &:nth-of-type(2) {
+      text-align: right;
+    }
+    &:nth-of-type(3) {
+      text-align: right;
+    }
+    &:nth-of-type(4) {
+      text-align: right;
+    }
+    &:nth-of-type(5) {
+      text-align: right;
+    }
+    &:nth-of-type(6) {
+      text-align: right;
+    }
+  `,
+}
 
-  const reportHeadingMap: { [name: string]: { heading: string; format?: string } } = {
-    timestamp: {
-      heading: t('earn.report.heading_timestamp'),
-    },
-    'cj amount/satoshi': {
-      heading: t('earn.report.heading_cj_amount'),
-      format: 'balance',
-    },
-    'my input count': {
-      heading: t('earn.report.heading_input_count'),
-    },
-    'my input value/satoshi': {
-      heading: t('earn.report.heading_input_value'),
-      format: 'balance',
-    },
-    'cjfee/satoshi': {
-      heading: t('earn.report.heading_cj_fee'),
-      format: 'balance',
-    },
-    'earned/satoshi': {
-      heading: t('earn.report.heading_earned'),
-      format: 'balance',
-    },
-    'confirm time/min': {
-      heading: t('earn.report.heading_confirm_time'),
-    },
-    'notes\n': {
-      heading: t('earn.report.heading_notes'),
-    },
+type Minutes = number
+
+interface EarnReportEntry {
+  timestamp: Date
+  cjTotalAmount: Api.AmountSats | null
+  inputCount: number | null
+  inputAmount: Api.AmountSats | null
+  fee: Api.AmountSats | null
+  earnedAmount: Api.AmountSats | null
+  confirmationDuration: Minutes | null
+  notes: string | null
+}
+
+// in the form of yyyy/MM/dd HH:mm:ss - e.g 2009/01/03 02:54:42
+type RawYielgenTimestamp = string
+
+const parseYieldgenTimestamp = (val: RawYielgenTimestamp) => {
+  // adding the timezone manually will display the date in the users timezone correctly
+  return new Date(Date.parse(`${val} GMT`))
+}
+
+const yieldgenReportLineToEarnReportEntry = (line: string): EarnReportEntry | null => {
+  if (!line.includes(',')) return null
+
+  const values = line.split(',')
+
+  // be defensive here - we cannot handle lines with unexpected values
+  if (values.length < 8) return null
+
+  return {
+    timestamp: parseYieldgenTimestamp(values[0]),
```

Comment:
I guess here we'd need to use the same format as we display to the user. Otherwise it becomes impossible to search for a date when you don't know what the underlying representation, that's actually being searched for, is.